### PR TITLE
Integrate hls.js with quality-levels

### PR DESCRIPTION
**Pull request by gkatsev**
Builds on top of https://github.com/muxinc/videojs-mux-kit/pull/63 and once that's merged in, this can be rebased and a PR against upstream be made.

### CODE REVIEW

**Comment by gkatsev** [comment]
The package-lock.json was regenerated because I used a newer version of npm. I can downgrade and regenerate it before we merge if that's an issue.

File: src/tech/hlsjs.js [hunk 1]
```diff
@@ -95,6 +101,91 @@ class HlsJs {
...
     }
   }
+
+  setupQualityLevels() {
+    const ql = this.player.qualityLevels();
+
+    let levels;
+
+    const lastIndex = () => {
+      var last;
```
Comment (luwes) [inline src/tech/hlsjs.js]:
```suggestion
      let last;
```

Comment (gkatsev) [reply src/tech/hlsjs.js]: ah, oops. This was left-over from when I was copying things into the chrome console.

**Review by luwes — COMMENTED** [review]
looks good to me, will leave it up to another reviewer for final approval

File: package.json [hunk 2]
```diff
@@ -44,6 +44,8 @@
     "sass-loader": "^10.2.0",
     "serve-handler": "^6.1.3",
     "style-loader": "^2.0.0",
+    "videojs-contrib-quality-levels": "^2.1.0",
+    "videojs-http-source-selector": "^1.1.6",
```
Comment (gkatsev) [inline package.json:48]: these two were added for the the quality-levels demo page.

File: webpack.dev.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ module.exports = async function(env) {
     entry: {
       index: [
         './src/entry.js',
+        'videojs-contrib-quality-levels',
+        'videojs-http-source-selector',
```
Comment (gkatsev) [inline webpack.dev.js:28]: easiest way to make the demo page work is to always import these two packages.

**Pull request closed without merging**